source/fab/application.py

```diff
@@ -58,6 +87,89 @@ def run(self, source: Path):
                 print('    found in: ' + str(filename))
                 print('    depends on: ' + str(fortran_db.depends_on(unit)))
 
+        # Start with the top level program unit
+        unit_to_process = [self._target]
+
+        # Initialise linker
+        if self._exec_name != "":
+            executable = Path(self._workspace) / self._exec_name
+        else:
+            executable = \
+                Path(self._workspace / self._target).with_suffix(".exe")
+
+        flags = self._command_flags_map.get(FortranLinker, [])
+        link_command = FortranLinker(self._workspace, flags, executable)
+
+        processed_units = []
+
+        while unit_to_process:
+            # Pop pending items from start of the list
+            unit = unit_to_process.pop(0)
+            if unit in processed_units:
+                continue
+
+            dependencies = fortran_db.depends_on(unit)
+            unit_to_process.extend(dependencies)
+
+            filenames = fortran_db.filenames_from_program_unit(unit)
+            # TODO: For now we simply can't handle this returning
+            #       multiple names; later we may be able to deal with
+            #       this via extra information in the configuration
+            if len(filenames) == 1:
+                filename = filenames[0]
+            else:
+                raise ValueError("Duplicate program unit found")
+
+            # Construct names of any expected module files to
+            # pass to the compiler constructor
+            mod_files = [Path(self._workspace /
+                              dependee).with_suffix('.mod')
```

Comment:
I think the simple way around this would probably be that the Compiler class will be expected to set an attribute that can be queried to indicate whether or not it produces `.mod` files?  Then you'd be able to check `if compiler_class.mod_files:` and then only do the above if it claims to produce them?  Either that or you just let the application pass them in this way and any Compiler class that doesn't need them can silently drop them?

Reply:
Given that we're not implementing any of this it doesn't matter. Just record it in a ticket and link that in this thread.

Reply:
Can do - #48